gemrb/core/Scriptable/Actor.cpp

```diff
@@ -6233,11 +6233,12 @@ bool Actor::ValidTarget(int ga_flags, const Scriptable *checker) const
 			if (Modified[IE_CHECKFORBERSERK]) return false;
 		}
 	}
-	if (ga_flags & GA_BUMPING) {
-		if (checker->Type != ST_ACTOR) return false;
-		if ((!((Actor*)(checker))->IsPartyMember() || this->GetStat(IE_EA) >= EA_EVILCUTOFF) && GetStat(IE_NPCBUMP) < 1) {
-			return false;
-		}
+	if (ga_flags & GA_ONLY_BUMPABLE) {
+		if (!checker) { return false; }
+		if (checker->Type != ST_ACTOR) { return false; }
+		if (GetStat(IE_NPCBUMP) < 1) return true;
+		if (!((Actor*)(checker))->IsPartyMember()) { return false; }
+		if (GetStat(IE_EA) >= EA_EVILCUTOFF) { return false; }
```

Comment:
This got moved from Map poorly. Only non-charmed party members and actors with IE_NPCBUMP can bump. Only neutrals and non-charmed party members can get bumped. 

So since this is checking if the actor can be bumped, the IE_NPCBUMP check doesn't belong here. And it's not enough to be in the party, you have to be non-charmed (controllable), eg. `IsPartyMember() && GetStat(IE_EA) < EA_GOODCUTOFF`.